### PR TITLE
Pull the resting screen code into separate class.

### PR DESCRIPTION
**Pull request by forslund**
The Mark-2 class does a lot of things, this encapsulates the resting screen code in a separate class in an attempt to make things more readable.

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -58,6 +58,86 @@ def clip(val, minimum, maximum):
...
+
+    def show(self):
+        """Show the idle screen or return to the skill that's overriding idle.
+        """
+        self.log.debug('Showing idle screen')
+        screen = None
+        if self.override_idle:
+            self.log.debug('Returning to override idle screen')
+            # Restore the page overriding idle instead of the normal idle
+            self.bus.semit(self.override_idle[0])
```
Comment (krisgesling) [inline __init__.py:119]: small typo

**Review by krisgesling — COMMENTED** [review]
Liking these changes, feels much cleaner to have them bundled up together. 

Seems to be working as expected with the small typo fix.

File: __init__.py [hunk 2]
```diff
@@ -98,7 +173,7 @@ def __init__(self):
         self.muted = False
         self.get_hardware_volume()       # read from the device
 
-        self.idle_override_set_time = time.monotonic()
+        self.resting = None
```
[the remark this answers is not in the document]
Comment (forslund) [reply __init__.py:176]: Sure I'll make that change.

File: __init__.py (hunk 1 above)
Comment (forslund) [reply __init__.py:119]: ah great catch

File: __init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (forslund) [reply __init__.py:193]: Change applied!

File: __init__.py (hunk 1 above)
Comment (forslund) [reply __init__.py]: Typo fixed

File: __init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (krisgesling) [reply __init__.py:193]: Github seems to be doing something funky.

I was just about to say that you missed this first one and suggest moving the `RestingScreen` instantiation to initialize, but that is what you've done in the branch and the commit is linked in the timeline below. 

The changes in this PR however are not showing that same code...

[the remark this answers is not in the document]
Comment (krisgesling) [reply __init__.py:193]: lol, nope apparently I just can't read code...

Seems like we can delete this first `self.resting` now

[the remark this answers is not in the document]
Comment (forslund) [reply __init__.py:193]: Doh! will fix

[the remark this answers is not in the document]
Comment (forslund) [reply __init__.py]: Now fixed :)

**Comment by krisgesling** [comment]
Looks great, thanks!

**Pull request merged**